lib/config/config-loader.js

```diff
@@ -173,26 +173,27 @@ async function loadConfigFile(filePath, allowTS) {
      */
     if (allowTS && isTS && !isDeno && !isBun) {
 
-        const createJiti = await import("jiti").then(jitiModule => (typeof jitiModule?.createJiti === "function" ? jitiModule.createJiti : jitiModule.default), () => {
+        // eslint-disable-next-line no-use-before-define -- `ConfigLoader.loadJiti` can be overwritten for testing
+        const { createJiti } = await ConfigLoader.loadJiti().catch(() => {
             throw new Error("The 'jiti' library is required for loading TypeScript configuration files. Make sure to install it.");
         });
 
+        // `createJiti` was added in jiti v2.
+        if (typeof createJiti !== "function") {
```

Comment:
If we are dropping support for v1, I think we should also update the version in package.json to `>=2`.

https://github.com/eslint/eslint/blob/064c8b612e2e4b773d6b25867f2045e3ceaa9d66/package.json#L201

Reply:
We purposely decided not to restrict the version range in order to avoid npm errors if a project has jiti installed in a version that's not compatible with ESLint. For example, if some other dependency requires an older version of jiti. The discussion can be found in https://github.com/eslint/eslint/pull/18134#discussion_r1689672788 in the PR where `jiti` was added.